internal/cli/cli.go

```diff
@@ -54,8 +68,23 @@ func (c *RootCommand) persistentPreRunE(cmd *cobra.Command, args []string) error
 }
 
 func (c *RootCommand) runE(cmd *cobra.Command, args []string) error {
-	log.Infof("To be implemented... log-level: %v", viper.GetString("log-level"))
-	return nil
+	// TODO get password from AWS ECR provider
+
+	buildah := buildah.Buildah{
+		Login: buildah.Login{
+			Registry: viper.GetString("registry"),
+			Username: "AWS",      // TODO use output from AWS ECR provider
+			Password: "password", // TODO use output from AWS ECR provider
+		},
+		Repo:     viper.GetString("repo"),
+		Commands: delimitByComma(viper.GetStringSlice("commands")),
```

Comment:
Should this be exposed directly? It seems like we would want a bit of a gated api interface here. If you configured manifests you are calling manifests. If you configure the build inputs you are calling build. 

It also seems odd that we are calling an ordered list of commands as user input.

Reply:
Good point, changed this to remove `commands` and have logic to add in commands based on args. The PR description has been updated with testing as you described.

Reply:
I think the commands we want to expose can be top level plugin settings keys. Those keys are the api we expose for each command.

instead of:
```
 settings:
    manifest_sources:
    - git-1
    - git-2
    manifest_targets:
    - git-1-2-combined
    - git-1-2-combined-2
```
something like:
```
 settings:
    manifest:
      sources:
      - git-1
      - git-2
      targets:
      - git-1-2-combined
      - git-1-2-combined-2
```

That would isolate the exposed apis to each command subkey in settings for simpler validation and ensure changes to an existing command or the addition of a new command would be independent of each other.

Reply:
Thanks, updated. The only caveat is that the input flags `version` and `manifest` now accept a JSON encoded string to match what Drone plugin would pass in via it's settings environment variable.

It's less intuitive than before with flat mapping but need to match behavior between CLI args and drone plugin env variable.